.husky/pre-commit

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+. "$(dirname $0)/_/husky.sh"
+
+git stash -q --keep-index
+
+# Add formatted files to commit
+files=$(npm run format | tail -n +5) || exit $?
+printf "$files" | while read line ; do
+  file="$(printf "$line" | cut -d' ' -f1)"
+  git add "$file"
+done;
```

Comment:
My shell scripting skills aren't that good, why not just:

```suggestion
# Format files and include formatting in commit
npm run format
git add .
```

The stashing on line 4 ensures non-staged files will not be included in the commit.

Reply:
@mondeja, I do think you need to stage the results of `npm run format` using e.g. `git add .`

Also, just out of curiosity, what is the advantage of having the ` || exit $?` suffixed? If it is to ensure the script fails if `npm run format` fails, I believe this is handled by the Husky script invoked at line 2, but might be worth testing this manually to be sure.

Reply:
> I believe this is handled by the Husky script invoked at line 2, but might be worth testing this manually to be sure.

Yes, I didn't know, removed :+1: